narwhals/_pandas_like/series.py

```diff
@@ -494,8 +500,10 @@ def sort(
         )
 
     def alias(self, name: str) -> Self:
-        ser = self._native_series
-        return self._from_native_series(ser.rename(name, copy=False))
+        if name != self.name:
+            ser = self._native_series
+            return self._from_native_series(ser.rename(name, copy=False))
```

Comment:
As discussed over the call, worth adding a note/warning in the `Series.alias` to explain what to expect, how to _not_ use this method for